runtime/j9vm/javanextvmi.cpp

```diff
@@ -358,6 +358,9 @@ JVM_VirtualThreadMountEnd(JNIEnv *env, jobject thread, jboolean firstMount)
 				goto release2;
 			}
 
+			/* Re-fetch as the memory allocation above may have moved the object. */
+			threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
```

Comment:
`j9jni_createGlobalRef` only allocates from a J9 Pool. `mmFuncs->J9AllocateObject` (line 341) is the GC trigger point. So, the re-fetch should move to line 346 after `J9AllocateObject` is invoked.